src/python/python-proto/python_proto/tests/strategies.py

```diff
@@ -33,16 +36,49 @@
     Label,
     MetricWrapper,
 )
-from python_proto.pipeline import Envelope, Metadata, Uuid
+from python_proto.pipeline import Envelope, Metadata, RawLog
+
+#
+# constants
+#
+# These values are used to parametrize the strategies defined below. Some of
+# them ensure the generated data actually makes sense. Others are used to
+# ensure strategies perform well. Please don't change these without a Very Good
+# Reason(TM).
+
 
 UINT64_MIN = 0
 UINT64_MAX = 2 ** 64 - 1
-INT64_MIN = -(2 ** 63) - 1
+
+INT64_MIN = -(2 ** 63) + 1
 INT64_MAX = 2 ** 63 - 1
+
+INT32_MIN = -(2 ** 31) + 1
+INT32_MAX = 2 ** 31 - 1
+
+# see https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#google.protobuf.Duration
+DURATION_SECONDS_MIN = -315_576_000_000
+DURATION_SECONDS_MAX = 315_576_000_000
```

Comment:
i so rarely see this lol

Reply:
I learned by accident here that this works in python as well, I know rust supports this kind of notation but had no idea python does too.